server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -3065,8 +3065,12 @@ public Network updateGuestNetwork(final UpdateNetworkCmd cmd) {
         }
 
         // network offering and domain suffix can be updated for Isolated networks only in 3.0
-        if ((networkOfferingId != null || domainSuffix != null) && network.getGuestType() != GuestType.Isolated) {
-            throw new InvalidParameterValueException("NetworkOffering and domain suffix upgrade can be performed for Isolated networks only");
+        if (networkOfferingId != null && network.getGuestType() != GuestType.Isolated) {
+            throw new InvalidParameterValueException("NetworkOffering update can be performed for Isolated networks only.");
+        }
+        // network offering and domain suffix can be updated for Isolated networks only in 3.0
+        if (domainSuffix != null && ! Arrays.asList(GuestType.Isolated, GuestType.Shared).contains(network.getGuestType())) {
```

Comment:
if domain suffix is changed, will VR be restarted to reflect the changes ?

Reply:
will check (code and life)

Reply:
code checked, this will work the same as with isolated networks. will test as well.